### ISSUE TITLE
layer generic typings

### ISSUE THREAD

**Issue by simonseyock** [hanreev/types-ol#26]
Hey, while using your project and also fiddling around with auto generating the type definitions directly from the openlayers source, I found some problems with the generic types of the layers.

I opened a PR for this here: https://github.com/hanreev/types-ol/pull/25

I fixed most of them already in the jsdoc typings (see openlayers/openlayers#11948) and I work on making the automatically generated typings better to begin with. If you have any suggestions what would be nice if it was already fixed upstream, let me know.

Also it would be really great if you could do a new release? In the current version on npm the ImageLayer and BaseImageLayer are broken.

**Comment by simonseyock** [comment]
I have to correct myself: the current release is not broken! I messed these files up myself somehow.

**Comment by simonseyock** [comment]
ah, by the way:

the problem with generic types is that something like this doesn't work:
```
const layerA: ImageLayer = new ImageLayer({ source: new ImageWMS() }):
const layerB: Layer<ImageWMS> = layerA;
```

with the PR this would be possible:
```
const layerA: ImageLayer<ImageWMS> = new ImageLayer<ImageWMS>({ source: new ImageWMS() }):
const layerB: Layer<ImageWMS> = layerA;
```

Also the `getSource` of layerA is then typed correctly and something like `layerA.getSource().getParams()` can be done.

**Comment by hanreev** [comment]
@simonseyock thanks for the feedback. i have published the update with these changes c2dd9f0
